src/components/BatteryIndicator/index.tsx

```diff
@@ -0,0 +1,107 @@
+import { useBattery } from 'hooks/battery';
+import styled from 'styled-components';
+
+const Container = styled.div`
+  width: fit-content;
+  height: 12px;
+  display: flex;
+  align-items: center;
+
+  --battery-healthy: #A5E07F;
+  --battery-warning: #D17F6B;
+`;
+
+const ChargeLevelContainer = styled.div`
+  height: 100%;
+  width: 20px;
+  position: relative;
+  display: flex;
+`;
+
+const ChargeLevel = styled.progress<{
```

Comment:
While the `progress` tag seems ideal here, it has historically poor browser compatibility along with a different implementation in each browser. Moreover,`-webkit-progress-bar` and `-webkit-progress-value` are only supported in, well, WebKit browsers 😅  So FireFox looks really funky since it has none of the styles defined in your `::-webkit` pseudo-elements:

<img width="224" alt="Screen Shot 2021-06-26 at 12 11 13 AM" src="https://user-images.githubusercontent.com/21055469/123505803-fd76a300-d615-11eb-8085-37e210134172.png">

So we should probably use an absolutely-positioned `div` approach to make the progress bar. I refactored this code a bit but maintained all your gradients, and this setup seems to have full compatibility with FireFox, Chrome, and Safari from my testing

```tsx
const ChargeLevelContainer = styled.div`
  height: 100%;
  width: 24px;
  position: relative;
  display: flex;
  border: 1px solid #626262ff;

  /* I moved the gray background gradient to this container */
  background: #54585b
    linear-gradient(180deg, transparent 25%, 65%, rgba(255, 255, 255, 0.5));
`;

const ChargeLevel = styled.div<{
  isCharging?: boolean;
  percent?: number;
}>`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;

  ${({ percent = 100, isCharging = false }) => css`
    width: ${percent}%;
    background: ${percent > 20 || isCharging
        ? batteryHealthyColor
        : batteryWarningColor}
      linear-gradient(
        180deg,
        transparent 10%,
        rgba(255, 255, 255, 0.5) 20%,
        35%,
        transparent 40%,
        rgba(0, 0, 0, 0.2) 55%,
        rgba(0, 0, 0, 0.3) 65%,
        rgba(0, 0, 0, 0.4) 80%
      );
  `};
`;
```

And then the JSX:
```tsx
<ChargeLevel percent={clampedChargePercent} isCharging={isCharging} />
```

Reply:
And here I thought I was scoring a totally rad semantic win by using `progress`! Yeah, I think you're right that the div approach is gonna be the way to go here after all.